python/cudf/cudf/core/column/categorical.py

```diff
@@ -657,17 +660,22 @@ def __setitem__(self, key, value):
 
     def _fill(
         self,
-        fill_value: ScalarLike,
+        fill_value: plc.Scalar,
```

Comment:
I just thought of this. How are you handling replacing `ScalarLike` with `plc.Scalar` for __public__ functions?

Reply:
At some point if a public API accepts a scalar, it will need to be annotated with `ScalarLike` i.e. "any scalar" regardless, but I think internal cudf functions should strive to accept/return a "typed" scalar like `plc.Scalar` or `pyarrow.Scalar` as much as possible for more predictability.

Reply:
Thanks, agreed.